### PR TITLE
Allow {posargs} in setenv

### PR DESCRIPTION
**Pull request by jayvdb**
Closes https://github.com/tox-dev/tox/issues/1695

## Contribution checklist:

(also see [CONTRIBUTING.rst](../tree/master/CONTRIBUTING.rst) for details)

- [x] wrote descriptive pull request text
- [x] added/updated test(s)
- [ ] updated/extended the documentation
- [x] added relevant [issue keyword](https://help.github.com/articles/closing-issues-using-keywords/)
      in message body
- [x] added news fragment in [changelog folder](../tree/master/docs/changelog)
  * fragment name: `<issue number>.<type>.rst` for example (588.bugfix.rst)
  * `<type>` is must be one of `bugfix`, `feature`, `deprecation`,`breaking`, `doc`, `misc`
  * if PR has no issue: consider creating one first or change it to the PR number after creating the PR
  * "sign" fragment with "by :user:`<your username>`"
  * please use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files - by :user:`superuser`."
  * also see [examples](../tree/master/docs/changelog)
- [x] added yourself to `CONTRIBUTORS` (preserving alphabetical order)

### CODE REVIEW

**Comment by jayvdb** [comment]
The changes re `{packages}` and `{opts}` doesnt radically improve them - they were errors (backtraces) before if they were used outside `install_command`, and they will still be backtraces, but now the error is the same as any other unknown substitution key.  As a result there isnt any tests in the PR for this, nor any changelog mention because the user doesnt see anything.  I'm happy to add it to the changelog entry anyway if it is strictly necessary to note any invisible/internal change.

**Review by gaborbernat — APPROVED** [review]

File: src/tox/config/__init__.py [hunk 1]
```diff
@@ -1771,11 +1801,8 @@ def _replace_match(self, match):
         if not any(g.values()):
             return os.pathsep
 
-        # special case: opts and packages. Leave {opts} and
-        # {packages} intact, they are replaced manually in
-        # _venv.VirtualEnv.run_install_command.
-        if sub_value in ("opts", "packages"):
```
Comment (jayvdb) [inline src/tox/config/__init__.py:1777]: This removal caused #1777

File: src/tox/config/__init__.py [hunk 2]
```diff
@@ -1883,28 +1912,17 @@ def getargvlist(cls, reader, value, replace=True):
...
         # shlex.
         if replace:
             newcommand = ""
             for word in CommandParser(command).words():
-                if word == "{posargs}" or word == "[]":
-                    newcommand += posargs_string
+                if word == "[]":
+                    newcommand += reader.getposargs()
                     continue
-                elif word.startswith("{posargs:") and word.endswith("}"):
```
Comment (jayvdb) [inline src/tox/config/__init__.py:1902]: This case wasn't handled correctly in the new code, if `:` was in the middle of this text range, resulting in regression https://github.com/tox-dev/tox/issues/1785

**Pull request merged**